cms/djangoapps/import_from_modulestore/tasks.py

```diff
@@ -0,0 +1,110 @@
+    except Import.DoesNotExist:
+        log.info('Import event not found for UUID %s', import_uuid)
+        return
+
+    import_event.clean_related_staged_content()
+    import_event.set_status(ImportStatus.STAGING)
+    try:
+        with transaction.atomic():
+            items_to_import = get_items_to_import(import_event)
+            for item in items_to_import:
+                staged_content = content_staging_api.stage_xblock_temporarily(
+                    item,
+                    import_event.user.id,
+                    purpose=IMPORT_FROM_MODULESTORE_STAGING_PURPOSE,
+                )
+                StagedContentForImport.objects.create(
+                    staged_content=staged_content,
+                    import_event=import_event,
+                    source_usage_key=item.location
+                )
+
+            if items_to_import:
+                import_event.set_status(ImportStatus.STAGED)
+            else:
+                import_event.set_status(ImportStatus.STAGING_FAILED)
+    except Exception as exc:  # pylint: disable=broad-except
+        import_event.set_status(ImportStatus.STAGING_FAILED)
+        raise exc
+
+
+@shared_task
+@set_code_owner_attribute
+def import_staged_content_to_library_task(
+    usage_keys_string: list[str],
+    import_uuid: str,
+    learning_package_id: int,
+    user_id: int,
+    composition_level: str,
+    override: bool,
+) -> None:
+    """
+    Import staged content to a library task.
+    """
+    validate_composition_level(composition_level)
+    try:
+        import_event = Import.objects.get(uuid=import_uuid, status=ImportStatus.STAGED, user_id=user_id)
+    except Import.DoesNotExist:
+        log.info('Ready import from modulestore not found')
+        return
+    try:
+        target_learning_package = LearningPackage.objects.get(id=learning_package_id)
+    except Import.LearningPackage:
+        log.info('Target learning package not found')
+        return
+
+    imported_publishable_versions = []
+    with authoring_api.bulk_draft_changes_for(learning_package_id=learning_package_id) as change_log:
+        try:
+            for usage_key_string in usage_keys_string:
+                if staged_content_item := import_event.get_staged_content_by_source_usage_key(usage_key_string):
+                    import_client = ImportClient(
+                        import_event,
+                        usage_key_string,
+                        target_learning_package,
+                        staged_content_item,
+                        composition_level,
+                        override,
+                    )
+                    imported_publishable_versions.extend(import_client.import_from_staged_content())
+        except Exception as exc:  # pylint: disable=broad-except
+            import_event.set_status(ImportStatus.IMPORTING_FAILED)
+            raise exc from exc  # TODO: retest raise and describe change_log commitment on Exception
```

Comment:
```suggestion
        except:  # pylint: disable=broad-except
            import_event.set_status(ImportStatus.IMPORTING_FAILED)
            raise
```
* You can use bare `raise` to re-raise the same exception that you caught
* `raise exc from exc` does not make sense-- that would be saying that "exc caused itself"
* Can you explain `TODO: retest raise and describe change_log commitment on Exception`? The `bulk_draft_changes_for` block will ensure that all DB changes are rolled back when an exception escapes. Is that what you wanted to make sure of?

Reply:
Refactored.

> Is that what you wanted to make sure of?

Yes, sorry, I didn't realize I hadn't removed this comment. 
It was a comment for myself to explore the behavior of the new `bulk_draft_changes_for` context manager when Exception raises.